runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkDoFnFunction.java

```diff
@@ -133,13 +135,23 @@ public void mapPartition(
 
   @Override
   public void open(Configuration parameters) throws Exception {
+    // Clear cache to get rid of any references to the Flink Classloader
+    // See https://jira.apache.org/jira/browse/BEAM-6460
+    PipelineOptionsFactory.resetCache();
```

Comment:
Note that I'm invalidating the cache when starting because the cache immediately reloads from the current classpath. So it doesn't make sense to do that during shutdown where the old classloader is still present.

We need a better solution in the long-term. This is just to unblock users.

Reply:
After some more investigation, I don't think this is necessary. The static cache will be removed automatically after the Beam job shuts down by unloading the PipelineOptionsFactory class.

The real problem here seems to be that the user code uses Flink's Jackson instance.